### PR TITLE
[libc] fix scudo integration build

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
FYI, scudo is using our header while not using our implementation.

### CODE REVIEW

**Review by jhuber6 — APPROVED** [review]

File: libc/test/integration/scudo/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,15 @@ add_entrypoint_library(
     libc.src.errno.errno
     libc.src.unistd.__llvm_libc_syscall
     libc.src.sched.__sched_getcpucount
+    libc.src.unistd.sysconf
```
[the remark this answers is not in the document]
Comment (SchrodingerZhu) [reply libc/test/integration/scudo/CMakeLists.txt]: done

**Review by lntue — APPROVED** [review]

**Comment by michaelrj-google** [comment]
Thanks for fixing this! At some point I want to address the comment I left in this file and make this a proper integration test, so I'll file an issue for that.

**Comment by SchrodingerZhu** [comment]
@nickdesaulniers I am not sure why it does not reproduce on your side, but it just reproduces and the debugging story starts there.

So I just attached the debugger the found out that it errors out on a failure of `mmap`. Using `strace`, I can see that the failed `mmap` is to apply a `MAP_FIXED` on an address that is not page-aligned. 

Then, it just comes to me that the allocator may get the wrong page size.

I re-run the test with debugger attached, and I tried `sysconf(1)` in the debugger (`_SC_PAGESIZE=1` in our header), and it returned a value that is apparently not correct. 

So, it must be the case that the implementation does not match the header. Indeed, the test can be fixed by enforcing the symbol implementation from our side.

**Comment by nickdesaulniers** [comment]
Ah! I was reproducing on the buildbot, and was attaching gdb to spot a malloc that would fail.

> Breakpoint 1, __GI___mmap64 (addr=0xffd7f7ee77ca, len=262144, prot=3, flags=50, fd=-1, offset=0)

I guess I didn't spot that `0xffd7f7ee77ca` is not a multiple of the page size and that `50` is `MAP_FIXED`.

Anyways, bravo, nice work!

**Pull request merged**